openlibrary/catalog/get_ia.py

```diff
@@ -79,9 +82,13 @@ def get_marc_record_from_ia(identifier):
     # Try marc.xml first
     if marc_xml_filename in filenames:
         data = urlopen_keep_trying(item_base + marc_xml_filename).read()
-        if data[:10].find('<?xml') != -1:
+        try:
```

Comment:
Out of curiosity, why is this wrapped in a try-catch block now?

Reply:
etree throws exceptions if the XML is not well formed. Previously the code was just looking for characters in a string. I haven't tested all possibilities, but I intended this to catch any unexpected server responses that aren't fully valid xml.

Reply:
Ah, ok. Could we catch an `etree.XMLSyntaxError` instead of a general exception? It looks like that's the only type of error it will throw.

Reply:
@cdrini I have tried to simplify things and removed that check for the `record` tag that you spotted would prevent collections from being loaded here and am leaving it to `MarcXml` to decide whether the input is valid. `MarcXml` already knew how to handle `collections` and didn't need the extra check.

 There are some `MarcXml` specific exceptions that can occur too, so I agree that being more specific is good, but there are a range of them :)

I tried removing the length check for Binary MARCs for the same logic that: `MarcBinary` should be able to handle its own input validation.... but discovered MarcBinary doesn't, and there are many binary MARCs with mismatched lengths, and that appears to be a good indicator of character encoding issues, so basically another big can of worms to sort out! :worried:

Reply:
@hornc Seems good. That does seem like an interesting way to detect wrongly encoded MARC files! If the lengths don't match, then it was probably converted to unicode at some point. Then we can try converting it from unicode to marc8, and if the length *is* what was expected, then that is very likely the case. But yeah, that can probably be the subject of another PR :P